controllers/placementapi_controller.go

```diff
@@ -1181,7 +1183,9 @@ func (r *PlacementAPIReconciler) ensureDeployment(
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 || *instance.Spec.Replicas == 0 {
+	// We need to check generation with a similar pattern to k8s,
+	// just in case when the status is recreated or the etcd database needs to be recreated
```

Comment:
I guess the comment does not apply any more as it is switched to ==

Reply:
done